desc/geometry/surface.py

```diff
@@ -469,6 +470,58 @@ def from_input_file(cls, path):
         )
         return surf
 
+    @classmethod
+    def from_near_axis(cls, aspect_ratio, elongation, mirror_ratio, axis_Z, NFP=1):
+        """Create a surface from a near-axis model for quasi-poloidal/quasi-isodynamic.
+
+        Parameters
+        ----------
+        aspect_ratio : float
+            Aspect ratio of the geometry = major radius / average cross-sectional area
+        elongation : float
+            Elongation of the elliptical surface = major axis / minor axis
+        mirror_ratio : float
+            Mirror ratio generated by toroidal variation of the cross-sectional area
+        axis_Z : float
+            Vertical extent of the magnetic axis Z coordinate; coefficient of sin(2*phi)
+        NFP : int
+            Number of field periods.
+
+        Returns
+        -------
+        surface : FourierRZToroidalSurface
+            Surface with given geometric properties.
+
+        """
+        a = np.sqrt(elongation) / aspect_ratio  # major axis
+        b = 1 / (aspect_ratio * np.sqrt(elongation))  # minor axis
+        epsilon = (2 - np.sqrt(4 - mirror_ratio ** 2)) / mirror_ratio
```

Comment:
should we include a note in the docstring that this needs mirror_ratio < 1?

Reply:
Yes that's a good idea.